zaza/openstack/charm_tests/heat/tests.py

```
@@ -0,0 +1,311 @@
+            env_path=None)
+
+        fields = {
+            'stack_name': STACK_NAME,
+            'timeout_mins': '15',
+            'disable_rollback': False,
+            'parameters': {
+                'admin_pass': 'Ubuntu',
+                'key_name': nova_utils.KEYPAIR_NAME,
+                'image': IMAGE_NAME
+            },
+            'template': template,
+            'files': dict(list(t_files.items()) + list(env_files.items())),
+            'environment': env
+        }
+
+        # Create the stack
+        try:
+            stack = self.heat_client.stacks.create(**fields)
+            logging.info('Stack data: {}'.format(stack))
+            stack_id = stack['stack']['id']
+            logging.info('Creating new stack, ID: {}'.format(stack_id))
+        except Exception as e:
+            # Generally, an api or cloud config error if this is hit.
+            msg = 'Failed to create heat stack: {}'.format(e)
+            logging.error(msg)
+            raise
+
+        # Confirm stack reaches COMPLETE status.
+        # /!\ Heat stacks reach a COMPLETE status even when nova cannot
+        # find resources (a valid hypervisor) to fit the instance, in
+        # which case the heat stack self-deletes!  Confirm anyway...
+        openstack_utils.resource_reaches_status(self.heat_client.stacks,
+                                                stack_id,
+                                                expected_status="COMPLETE",
+                                                msg="Stack status wait")
+        # List stack
+        stacks = list(self.heat_client.stacks.list())
+        logging.info('All stacks: {}'.format(stacks))
+
+        # Get stack information
+        try:
+            stack = self.heat_client.stacks.get(STACK_NAME)
+        except Exception as e:
+            # Generally, a resource availability issue if this is hit.
+            msg = 'Failed to get heat stack: {}'.format(e)
+            assert False, msg
```

Either drop the try/except (and let the exception just fail the test), or replace this bit with:

```python
self.fail('Failed to get ...')
```